internal/cmd/gencli/templates.go

```diff
@@ -0,0 +1,466 @@
+	init{{ camelCase .SubActionPrefix }}Flags()
+	
+	var helpStr string
+	helpMap := common.HelpMap("{{ lowerSpaceCase .ResourceType }}")
+
+	switch c.Func {
+	{{ if not .SkipNormalHelp }}
+	{{ range $i, $action := .StdActions }}
+	case "{{ $action }}":
+		helpStr = helpMap[c.Func]() + c.Flags().Help()
+	{{ end }}
+	{{ end }}
+	default:
+	{{ if .HasExtraHelpFunc }}
+		helpStr = c.extra{{ camelCase .SubActionPrefix }}HelpFunc(helpMap)
+	{{ else }}
+		helpStr = helpMap["base"]()
+	{{ end }}
+	}
+
+	// Keep linter from complaining if we don't actually generate code using it
+	_ = helpMap
+	return helpStr
+}
+
+var flags{{ camelCase .SubActionPrefix }}Map = map[string][]string{
+	{{ range $i, $action := .StdActions }}
+	{{ if eq $action "create" }}
+	"create": { "{{ kebabCase $input.Container }}-id", "name", "description" },
+	{{ end }}
+	{{ if eq $action "read" }}
+	"read": {"id"},
+	{{ end }}
+	{{ if eq $action "update" }}
+	"update": {"id", "name", "description" {{ if hasAction $input.VersionedActions "update" }}, "version" {{ end }} },
+	{{ end }}
+	{{ if eq $action "delete" }}
+	"delete": {"id"},
+	{{ end }}
+	{{ if eq $action "list" }}
+	"list": { "{{ kebabCase $input.Container }}-id" {{ if (eq $input.Container "Scope") }}, "recursive"{{ end }} },
+	{{ end }}
+	{{ end }}
+}
+
+func (c *{{ camelCase .SubActionPrefix }}Command) Flags() *base.FlagSets {
+	if len(flags{{ camelCase .SubActionPrefix }}Map[c.Func]) == 0 {
+		return c.FlagSet(base.FlagSetNone)
+	}
+
+	set := c.FlagSet(base.FlagSetHTTP | base.FlagSetClient | base.FlagSetOutputFormat)
+	f := set.NewFlagSet("Command Options")
+	common.PopulateCommonFlags(c.Command, f, "{{ if .SubActionPrefix }}{{ .SubActionPrefix }}-type {{ end }}{{ lowerSpaceCase .ResourceType }}", flags{{ camelCase .SubActionPrefix }}Map[c.Func])
+
+	extra{{ camelCase .SubActionPrefix }}FlagsFunc(c, set, f)
+
+	return set
+}
+
+func (c *{{ camelCase .SubActionPrefix }}Command) Run(args []string) int {
+	init{{ camelCase .SubActionPrefix }}Flags()
+
+	{{ if .HasExampleCliOutput }}
+	if os.Getenv("BOUNDARY_EXAMPLE_CLI_OUTPUT") != "" {
+		c.UI.Output(exampleOutput())
+		return 0
+	}
+	{{ end }}
+
+	switch c.Func {
+	case "":
+		return cli.RunResultHelp
+	{{ if .IsAbstractType }}
+	case "create", "update":
+		return cli.RunResultHelp
+	{{ end }}
+	}
+
+	c.plural = "{{ if .SubActionPrefix }}{{ .SubActionPrefix }}-type {{ end }}{{ lowerSpaceCase .ResourceType }}"
+	switch c.Func {
+	case "list":
+		c.plural = "{{ if .SubActionPrefix }}{{ .SubActionPrefix }}-type {{ end }}{{ lowerSpaceCase .ResourceType }}s"		
+	}
+
+	f := c.Flags()
+
+	if err := f.Parse(args); err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+
+	{{ if .HasId }}
+	if strutil.StrListContains(flags{{ camelCase .SubActionPrefix }}Map[c.Func], "id") && c.FlagId == "" {
+			c.UI.Error("ID is required but not passed in via -id")
+			return 1
+	}
+	{{ end }}
+
+	var opts []{{ .Pkg }}.Option
+
+	{{ if .Container }}
```

Comment:
Good point. Originally I was going to have Container be an _override_ for it not to be Scope, but ended up just making it always specified. I could go either way, but for now I'll remove the `if` in the template since simplifying the template is probably better than simplifying the input structs.